webrender/src/device.rs

```diff
@@ -1629,6 +1619,166 @@ impl<B: hal::Backend> RenderPass<B> {
     }
 }
 
+pub struct DescPool<B: hal::Backend> {
+    descriptor_pool: B::DescriptorPool,
+    descriptor_set: Vec<B::DescriptorSet>,
+    descriptor_set_layout: B::DescriptorSetLayout,
+    current_descriptor_set_id: usize,
+    max_descriptor_set_size: usize,
+}
+
+impl<B: hal::Backend> DescPool<B> {
+    pub fn new(
+        device: &B::Device,
+        max_size: usize,
+        descriptor_range_descriptors: Vec<DescriptorRangeDesc>,
+        descriptor_set_layout: Vec<DescriptorSetLayoutBinding>,
+    ) -> Self {
+        let descriptor_pool =
+            device.create_descriptor_pool(
+                max_size,
+                descriptor_range_descriptors.as_slice(),
+            );
+        let descriptor_set_layout = device.create_descriptor_set_layout(&descriptor_set_layout);
+        let mut dp = DescPool {
+            descriptor_pool,
+            descriptor_set: vec!(),
+            descriptor_set_layout,
+            current_descriptor_set_id: 0,
+            max_descriptor_set_size: max_size,
+        };
+        dp.next();
+        dp.reset();
+        dp
+    }
+
+    pub fn get_descriptor_set(&mut self) -> &B::DescriptorSet {
```

Comment:
nit: the Rust API guidelines say that simple accessors shouldn't have "get_" prefix